### PR TITLE
Users/groups/roles swagger docs

### PR DESCRIPTION
**Pull request by NielsCharlier**

### CODE REVIEW

File: doc/en/api/1.0.0/usergroup.yaml [hunk 1]
```diff
@@ -0,0 +1,378 @@
...
+          required: true
+          type: string
+      responses:
+        200:
+          description: OK
+          schema:
+            $ref: "#/definitions/User"  
+           
+    post:
+      operationId: userPost
```
Comment (tbarsballe) [inline doc/en/api/1.0.0/usergroup.yaml]: operationId should be unique (same as line 109)

File: doc/en/api/1.0.0/usergroup.yaml [hunk 2]
```diff
@@ -0,0 +1,378 @@
...
+          in: body
+          required: true
+          schema:
+            $ref: "#/definitions/User" 
+      responses:
+        201:
+          description: Created
+          headers:
+            ID:
+              description: ID of the newly created user
```
Comment (tbarsballe) [inline doc/en/api/1.0.0/usergroup.yaml]: Schema property should have a `type`. `type: string` is fine.

File: doc/en/api/1.0.0/usergroup.yaml [hunk 3]
```diff
@@ -0,0 +1,378 @@
...
+            $ref: "#/definitions/User"    
+      consumes: 
+        - application/json
+        - application/xml
+      responses:
+        201:
+          description: Created
+          headers:
+            ID:
+              description: ID of the newly created user
```
Comment (tbarsballe) [inline doc/en/api/1.0.0/usergroup.yaml]: Schema property should have a `type`. `type: string` is fine.

File: doc/en/api/1.0.0/usergroup.yaml [hunk 4]
```diff
@@ -0,0 +1,378 @@
...
+      operationId: groupPost
+      summary: Add a group
+      description: Add a group in the default user/group service.
+      consumes: 
+        - application/json
+        - application/xml
+      parameters:
+        - name: serviceName
+          description: the name of the user/group service
+          in: path
```
Comment (tbarsballe) [inline doc/en/api/1.0.0/usergroup.yaml]: serviceName doesn't show up in the path, is this an error?

File: doc/en/api/1.0.0/usergroup.yaml [hunk 5]
```diff
@@ -0,0 +1,378 @@
...
+          description: the name of the group
+          in: path
+          required: true
+          type: string
+      responses:
+        200:
+          description: OK
+  /usergroup/service/{serviceName}/groups/{group}:
+    post:
+      operationId: groupPost
```
Comment (tbarsballe) [inline doc/en/api/1.0.0/usergroup.yaml]: operationIds should be unique, this is the same as on line 257

File: doc/en/api/1.0.0/usergroup.yaml [hunk 6]
```diff
@@ -0,0 +1,378 @@
...
+      operationId: groupDefaultDelete
+      summary: Delete a group
+      description: Delete a group in the default user/group service.
+      consumes: 
+        - application/json
+        - application/xml
+      parameters:
+        - name: serviceName
+          description: the name of the user/group service
+          in: path
```
Comment (tbarsballe) [inline doc/en/api/1.0.0/usergroup.yaml]: serviceName doesn't show up in the path, is this an error?

File: doc/en/api/1.0.0/roles.yaml [hunk 7]
```diff
@@ -0,0 +1,143 @@
...
+      operationId: rolePost
+      summary: Add a role
+      description: Add a role in the default role service.
+      consumes: 
+        - application/json
+        - application/xml
+      parameters:
+        - name: serviceName
+          description: the name of the role service
+          in: path
```
Comment (tbarsballe) [inline doc/en/api/1.0.0/roles.yaml]: serviceName doesn't show up in the path, is this an error?

File: doc/en/api/1.0.0/roles.yaml [hunk 8]
```diff
@@ -0,0 +1,143 @@
...
+      operationId: roleDefaultDelete
+      summary: Delete a role
+      description: Delete a role in the default role service.
+      consumes: 
+        - application/json
+        - application/xml
+      parameters:
+        - name: serviceName
+          description: the name of the role service
+          in: path
```
Comment (tbarsballe) [inline doc/en/api/1.0.0/roles.yaml]: serviceName doesn't show up in the path, is this an error?

File: doc/en/api/1.0.0/roles.yaml [hunk 9]
```diff
@@ -0,0 +1,143 @@
...
+          description: the name of the role
+          in: path
+          required: true
+          type: string
+      responses:
+        200:
+          description: OK
+  /roles/service/{serviceName}/roles/{role}:
+    post:
+      operationId: rolePost
```
Comment (tbarsballe) [inline doc/en/api/1.0.0/roles.yaml]: operationIds must be unique, this is the same as line 45

File: doc/en/api/1.0.0/roles.yaml [hunk 10]
```diff
@@ -0,0 +1,143 @@
+---
+swagger: '2.0'
+info:
+  version: 1.0.0
+  title: Users, Groups and Roles
```
Comment (tbarsballe) [inline doc/en/api/1.0.0/roles.yaml]: Should this be just "Roles"? Description also need updating

File: doc/en/api/1.0.0/usergroup.yaml [hunk 11]
```diff
@@ -0,0 +1,378 @@
+---
+swagger: '2.0'
+info:
+  version: 1.0.0
+  title: Users, Groups and Roles
```
Comment (tbarsballe) [inline doc/en/api/1.0.0/usergroup.yaml]: Should this just be "Users and Groups"? Description also needs updating.

**Comment by NielsCharlier** [comment]
@tbarsballe thanks for the review I made the changes

**Comment by tbarsballe** [comment]
Changes look good, merging. Thanks for the docs.

**Pull request merged**